### PR TITLE
docs: update `middleware.mdx` doc about revalidation issue

### LINKED ISSUE #846: [Docs]: Revalidation on translated path
### Link to page

_No response_

### Describe the problem

Hi,

Versions concerned :
```
"next": "14.1.0",
"next-intl": "3.5.4",
```

I'm using the [localized pathnames strategy ](https://next-intl-docs.vercel.app/docs/routing/navigation#localized-pathnames) to do the routing. Here's an example : 
```
.
└── [locale]/
    └── news/
        └── [slug]
```
In `navigation.ts`, localized path is set like this : 
```js
{
  '/news/[slug]': {
    en: '/news/[slug]',
    fr: '/infos/[slug]',
    it: '/notizia/[slug]',
  },
}
```

Theses pages are generated at build time thanks to [`generateStaticParams()`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params). But, when I'm trying to revalidate the page with [`revalidatePath('/fr/infos/some-url')`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) it doesn't work.

After a lot of investigation, I came to this result :

For pages that are statically generated : 
- This doesn't work : `revalidatePath('/fr/infos/some-url')`
- This does work : `revalidatePath('/fr/news/some-url')`

For pages that are dynamic : 
- This doesn't work : `revalidatePath('/fr/news/some-url')`
- This does work : `revalidatePath('/fr/infos/some-url')`

I found this by looking at the `prerender-manifest.json` generated after a Next's build. I tested to revalidate the `dataRoute` and it worked.
```json
{
    "/fr/news/article-french": {
      "experimentalBypassFor": [
        { "type": "header", "key": "Next-Action" },
        {
          "type": "header",
          "key": "content-type",
          "value": "multipart/form-data"
        }
      ],
      "initialRevalidateSeconds": false,
      "srcRoute": "/[locale]/news/[slug]",
      "dataRoute": "/fr/news/article-french.rsc"
    },
}
```

Maybe it's worth to add something in the doc about how revalidation is working when we are using localized folder names ?
Or maybe there is a fix to do in code ?

### PR DESCRIPTION
**Pull request by LouisCuvelier**
Hey,

As requested [here](https://github.com/amannn/next-intl/issues/846), I've update the doc about the ongoing revalidation issue

Fixes #846

### CODE REVIEW

File: docs/pages/docs/routing/middleware.mdx [hunk 1]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
   APIs](/docs/routing/navigation#localized-pathnames) in your components.
 </Callout>
 
+<details>
+<summary>How to revalidate localized pathnames?</summary>
+
+Depending on how the route is generated, whether statically or dynamically, if you're attempting to use Next.js's function `revalidatePath()`, you will need to call it with two different pathnames.
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:349]:
```suggestion
Depending on if a route is generated statically (at build time) or dynamically (at runtime), [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) needs to be called either for the localized or the internal pathname.
```

File: docs/pages/docs/routing/middleware.mdx [hunk 2]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
...
+
+Let's take an example with this folder structure:
+```
+.
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:353]:
```suggestion
app
```

File: docs/pages/docs/routing/middleware.mdx [hunk 3]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
...
+└── [locale]/
+    └── news/
+        └── [slug]
+```
+
+And, this localized configuration :
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:359]:
```suggestion
… and this middleware configuration:

```

File: docs/pages/docs/routing/middleware.mdx [hunk 4]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
...
+import createMiddleware from 'next-intl/middleware';
+ 
+export default createMiddleware({
+  defaultLocale: 'en',
+  locales: ['en', 'fr'],
+  pathnames: {
+    '/news/[slug]': {
+      en: '/news/[slug]',
+      fr: '/infos/[slug]'
+    },
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:370]:
```suggestion
    }
```

File: docs/pages/docs/routing/middleware.mdx [hunk 5]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
...
+  }
+});
+```
+
+Assuming the pathname is `/fr/infos/some-article`, if the page is dynamically generated, you will need to revalidate like this: `revalidatePath('/fr/infos/some-article')`. However, if the page is statically rendered, the path to revalidate is this one: `revalidatePath('/fr/news/some-article')`. So, if you don't know if the page is statically or dynamically generated, we advise you to revalidate both.
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:375]: Good point to revalidate both when in doubt 👍 

```suggestion
Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can invalidate the route like this:

```tsx
// Statically generated at build time
revalidatePath('/fr/news/some-article')

// Dynamically generated at runtime:
revalidatePath('/fr/infos/some-article')
```-

When in doubt, you can revalidate both paths to be on the safe side.
```

File: docs/pages/docs/routing/middleware.mdx [hunk 6]
```diff
@@ -343,6 +343,42 @@ export default createMiddleware({
...
+
+You can find more information on the issue [here](https://github.com/amannn/next-intl/issues/846).
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:377]: Maybe we can link to the Next.js issue? I'd like to close the issue on the `next-intl` side once your PR here is merged.

```suggestion
See also [`vercel/next.js#59825`](https://github.com/vercel/next.js/issues/59825).
```

**Review by amannn — COMMENTED** [review]
Thanks for following up and making your solution available to others!

I have left a few potential wording improvements, let me know if these sound right to you!

**Comment by LouisCuvelier** [comment]
Thanks for the corrections. Ready to merge ;)

File: docs/pages/docs/routing/middleware.mdx [hunk 7]
```diff
@@ -343,6 +343,53 @@ export default createMiddleware({
...
+<details>
+<summary>How to revalidate localized pathnames?</summary>
+
+Depending on if a route is generated statically (at build time) or dynamically (at runtime), [`revalidatePath`](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) needs to be called either for the localized or the internal pathname.
+
+Let's take an example with this folder structure:
+```
+app
+└── [locale]/
+    └── news/
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:355]:
```suggestion
└── [locale]
    └── news
```

File: docs/pages/docs/routing/middleware.mdx [hunk 8]
```diff
@@ -343,6 +343,53 @@ export default createMiddleware({
   APIs](/docs/routing/navigation#localized-pathnames) in your components.
 </Callout>
 
+<details>
+<summary>How to revalidate localized pathnames?</summary>
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:347]:
```suggestion
<summary>How can I revalidate localized pathnames?</summary>
```

File: docs/pages/docs/routing/middleware.mdx [hunk 9]
```diff
@@ -343,6 +343,53 @@ export default createMiddleware({
...
+  pathnames: {
+    '/news/[slug]': {
+      en: '/news/[slug]',
+      fr: '/infos/[slug]'
+    }
+  }
+});
+```
+
+Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can invalidate the route like this:
```
Comment (amannn) [inline docs/pages/docs/routing/middleware.mdx:376]:
```suggestion
Depending on whether `some-article` was included in [`generateStaticParams`](https://nextjs.org/docs/app/api-reference/functions/generate-static-params) or not, you can revalidate the route like this:
```

**Review by amannn — COMMENTED** [review]
Another round of small wording improvements, but ready to merge now from my side too!

Thank you so much @LouisCuvelier! 🙌

**Pull request merged**